engine/explain.go

```diff
@@ -61,13 +61,25 @@ func (a *AnalyzeOutputNode) aggregateSamples() {
 		for _, child := range a.Children {
 			childPeak := child.PeakSamples()
 			a.peakSamples = max(a.peakSamples, childPeak)
+
+			if a.OperatorTelemetry.SubQuery() {
+				// Skip aggregating from subquery to avoid double counting samples from children.
+				continue
+			}
+			if a.OperatorTelemetry.StepInvariant() {
+				// Children of step invariant operator outputs one step, but they should be counted towards all the steps.
+				for i := 0; i < len(a.totalSamplesPerStep); i++ {
+					childSamples := child.TotalSamples()
```

Comment:
Do we need to recalculate this for each loop iteration?

I also wonder if we can modify the implementation of the step invariant analysis to fit into the existing algorithm instead of tweaking the algorithm to specially handle the step invariant case.

Reply:
> I also wonder if we can modify the implementation of the step invariant analysis

The other option is to IncrementSamples() in StepInvariant operator and skip counting samples from its children. I think I prefer this option.